components/script/dom/location.rs

```diff
@@ -413,13 +413,14 @@ impl LocationMethods<crate::DomTypeHolder> for Location {
         if self.has_document() {
             // Note: no call to self.check_same_origin_domain()
             // Step 2: Parse the given value relative to the entry settings object.
-            // If that failed, throw a TypeError exception.
+            // Step 3: If url is failure, then throw a "SyntaxError" DOMException.
             let base_url = self.entry_settings_object().api_base_url();
             let url = match base_url.join(&value.0) {
                 Ok(url) => url,
-                Err(e) => return Err(Error::Type(format!("Couldn't parse URL: {}", e))),
+                // TODO: include details?
+                Err(_e) => return Err(Error::Syntax),
```

Comment:
This is problem with our error types as they do not support writing any messages.

Reply:
I opened #39053 about this.